src/features/diagnosticsProvider.ts

```diff
@@ -319,36 +324,59 @@ class DiagnosticsProvider extends AbstractSupport {
         });
     }
 
-    private static _shouldInclude(quickFix: protocol.QuickFix): boolean {
-        const config = vscode.workspace.getConfiguration('csharp');
-        if (config.get('suppressHiddenDiagnostics', true)) {
-            return quickFix.LogLevel.toLowerCase() !== 'hidden';
-        } else {
-            return true;
+    private _asDiagnostic(quickFix: protocol.QuickFix): vscode.Diagnostic | undefined {
+        // CS0162 & CS8019 => Unnused using and unreachable code.
+        // These hard coded values bring some goodnes of fading even when analyzers are disabled.
+        let isFadeout = (quickFix.Tags && !!quickFix.Tags.find(x => x.toLowerCase() == 'unnecessary')) || quickFix.Id == "CS0162" || quickFix.Id == "CS8019";
+
+
+        let severity = this._asDiagnosticSeverity(quickFix, isFadeout);
+
+        if(severity === undefined)
+        {
+            return undefined;
         }
-    }
 
-    // --- data converter
+        let message = `${quickFix.Text} [${quickFix.Projects.map(n => this._asProjectLabel(n)).join(', ')}]`;
 
-    private static _asDiagnostic(quickFix: protocol.QuickFix): vscode.Diagnostic {
-        let severity = DiagnosticsProvider._asDiagnosticSeverity(quickFix.LogLevel);
-        let message = `${quickFix.Text} [${quickFix.Projects.map(n => DiagnosticsProvider._asProjectLabel(n)).join(', ')}]`;
-        return new vscode.Diagnostic(toRange(quickFix), message, severity);
+        let diagnostic = new vscode.Diagnostic(toRange(quickFix), message, severity);
+
+        if(isFadeout)
```

Comment:
Can we refactor this a little to handle all the fadeout special stuff in one place?